src/roiextractors/extractors/tiffimagingextractors/scanimagetiffimagingextractor.py

```diff
@@ -22,13 +24,487 @@ def _get_scanimage_reader() -> type:
     ).ScanImageTiffReader
 
 
-class ScanImageTiffImagingExtractor(ImagingExtractor):
+def extract_extra_metadata(
+    file_path,
+) -> dict:  # TODO: Refactor neuroconv to reference this implementation to avoid duplication
+    """Extract metadata from a ScanImage TIFF file.
```

Comment:
I would suggest a method name and docstring addition specifying what versions this is known to support for reading, similar to the v3_8 below